### DISCUSSION TITLE
Function issua

### DISCUSSION THREAD

**Discussion by MuffinHavlop** [MastodonRoboticsTeam/VRC2023#13] [Q&A] [answered]
How can I program a button to turn a component on and keep it running and then turns off until I push the same button the second time

**Comment by ghost** [comment, marked as answer]
There are two ways:

First way: If you press the button it will run the servo functions , else the servo functions will set servo's speed to 0

```
if (ps2.Button(PSB_R1)) {
   ctrl_servo360(SRV_INTAKE, SPD_INTAKE);
}
if (ps2x.ButtonReleased(PSB_L1) {
   ctrl_servo360(SRV_INTAKE, 0);
}
```

Second way:  use boolean to control the flow of servos.


```
  bool intake_toggle = false;
  bool intake = false;

   if (ps2.Button(PSB_L1)){
      if(!intake_toggled) {
        intake_toggled = true;
        intake = !intake;
        ctrl_servo360(SRV_INTAKE, (intake) ?  SPD_INTAKE : 0);
    } else intake_toggle = false;


```
